modules/javafx.graphics/src/main/native-glass/mac/GlassView3D.m

```diff
@@ -93,7 +93,7 @@ - (CGLPixelFormatObj)_createPixelFormatWithDepth:(CGLPixelFormatAttribute)depth
         };
         GLint npix = 0;
         CGLError err = CGLChoosePixelFormat(attributes, &pix, &npix);
-        if ((err == kCGLNoError) && (npix == 0))
+        if (pix == NULL)
         {
```

Comment:
Could you please be more specific on why you suggest the check be expended with  `|| err != kCGLNoError`?
I see no harm in adding it but it feels like an unnecessary condition to me; my understanding of the current [documentation](https://developer.apple.com/library/archive/documentation/GraphicsImaging/Conceptual/OpenGL-MacProgGuide/opengl_pixelformats/opengl_pixelformats.html) (though the use of the API itself is deprecated by Apple) has me understand than `pix == NULL` is both a necessary and sufficient test.

Reply:
From the documentation, the check `pix == NULL` seems sufficient, but the `err != kCGLNoError` was used before, so I just want to keep it safe. If the issue occurs without error getting printed it will be difficult to trace. As you observed the OpenGL is deprecated, so suspecting documentation may not be up to date. Also this [doc](https://developer.apple.com/library/mac/documentation/GraphicsImaging/Reference/CGL_OpenGL/#//apple_ref/c/func/CGLChoosePixelFormat) was referred for fixing  [JDK-8154148](https://bugs.openjdk.java.net/browse/JDK-8154148), and is no longer available.
However I can't find any documentation to support this `err != kCGLNoError` check.

Reply:
Actually, the existing check uses `err == kCGLNoError` which seems backwards (and likely is a big part of why this bug is happening). My recommendation is to not add in the check for `err != kCGLNoError`, and just use `if (pix == NULL)`.

The suggestion to add the print statement is a good one.

Reply:
Actually, `err == kCGLNoError` was used before and not `err != kCGLNoError`  (i.e. equals and not different) and that is in fact the reason for the bug (in conjunction with the AND).
As you note, it's unfortunate that the link you mentioned is no longer valid as it would be very useful to understand the motivation behind the original fix in JDK-8154148, because it is indeed quite puzzling.

Please note that in any case a check on `err != kCGLNoError` is done just before the function returns and the error is logged if true.

Reply:
OK, so we're in agreement on the fix itself. Good.

The existing logging done right before the function returns will indeed log a complete failure to find a valid pixel format. It won't, however, log the fact that the initial attempt failed and we then try to use a fallback. It was this that @arapte asked to be logged. I agree with his request.

Reply:
It looks good then, please update PR without `err != kCGLNoError`

Reply:
Something like this, maybe?

```
    if (pix == NULL) {
        NSLog(@"CGLChoosePixelFormat: unable to find a pixel format, trying again with reduced capabilities");
```

The message could perhaps be improved, but something indicating we are using a fallback would be helpful.

Reply:
> The existing logging done right before the function returns will indeed log a complete failure to find a valid pixel format. It won't, however, log the fact that the initial attempt failed and we then try to use a fallback. It was this that @arapte asked to be logged. I agree with his request.

Agreed. 
I suggest the printout only explains that we failed to get a pixel format object and we're about to try again but doesn't mention the returned error; since we don't check on its value, we could end up printing something like "Failed to do something very important[..] (Error: No error)", which would just be confusing IMO.